File: 232-用栈实现队列.cpp
class MyQueue {
public:
    stack<int> stack_enqueue;
    stack<int> stack_dequeue;
    int a;

    /** Initialize your data structure here. */
    MyQueue() {

    }
    
    /** Push element x to the back of queue. */
    void push(int x) {
        stack_enqueue.push(x);
    }
    
    /** Removes the element from in front of queue and returns that element. */
    int pop() {
        if(stack_dequeue.empty()){
            while(!stack_enqueue.empty()){
                stack_dequeue.push(stack_enqueue.top());
                stack_enqueue.pop();
            }
        }
        a = stack_dequeue.top();
        stack_dequeue.pop();
        return a;
    }
    
    /** Get the front element. */
    int peek() {
        if(stack_dequeue.empty()){
            while(!stack_enqueue.empty()){
                stack_dequeue.push(stack_enqueue.top());
                stack_enqueue.pop();
            }
        }
        return stack_dequeue.top();
    }
    
    /** Returns whether the queue is empty. */
    bool empty() {
        return stack_dequeue.empty() && stack_enqueue.empty();
    }
};

File: 232-用栈实现队列.py
class MyQueue:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.stack_enqueue = []
        self.stack_dequeue = []

    def push(self, x: int) -> None:
        """
        Push element x to the back of queue.
        """
        self.stack_enqueue.append(x)

    def pop(self) -> int:
        """
        Removes the element from in front of queue and returns that element.
        """
        if not self.stack_dequeue:
            while self.stack_enqueue:
                self.stack_dequeue.append(self.stack_enqueue.pop())
        return self.stack_dequeue.pop()

        

    def peek(self) -> int:
        """
        Get the front element.
        """
        if not self.stack_dequeue:
            while self.stack_enqueue:
                self.stack_dequeue.append(self.stack_enqueue.pop())
        return self.stack_dequeue[-1]
        

    def empty(self) -> bool:
        """
        Returns whether the queue is empty.
        """
        return not self.stack_enqueue and not self.stack_dequeue
